hadoop-ozone/ozonefs-common/src/main/java/org/apache/hadoop/fs/ozone/BasicRootedOzoneClientAdapterImpl.java

```diff
@@ -882,9 +882,9 @@ public List<FileStatusAdapter> listStatus(String pathStr, boolean recursive,
     }
     OFSPath ofsStartPath = new OFSPath(startPath, config);
     if (ofsPath.isVolume()) {
-      String startBucket = ofsStartPath.getBucketName();
+      String startBucketPath = ofsStartPath.getNonKeyPath();
       return listStatusVolume(ofsPath.getVolumeName(),
-          recursive, startBucket, numEntries, uri, workingDir, username);
+          recursive, startBucketPath, numEntries, uri, workingDir, username);
```

Comment:
I suggest changing `startPath` parameter of `listStatusVolume` to `OFSPath`, and passing `ofsStartPath`.  This reduces the chance for such misuse, and also avoids creating another instance of `OFSPath` for the same path in the method.  The method may also validate that `ofsStartPath.getVolumeName()` equals `volumeNameStr`.

Reply:
Currently, all the following methods are using String as parameter, maybe we can align to use String here or it might confuse people more?
```
listStatusBucketSnapshot(ofsPath.getVolumeName(), ofsPath.getBucketName(), uri);
listStatusVolume(ofsPath.getVolumeName(), recursive, startBucketPath, numEntries, uri, workingDir, username);
listStatusRoot(recursive, startPath, numEntries, uri, workingDir, username);
```